src/edges_analysis/calibration/calibrate.py

```diff
@@ -495,19 +497,23 @@ def apply_beam_correction(
     new_data = data.data.copy()
 
     resids = data.residuals.copy() if data.residuals is not None else None
-    for i, (lst0, lst1) in enumerate(data.lst_ranges[:, 0, :]):
-        new = beam.between_lsts(lst0.hour, lst1.hour)
-        if integrate_before_ratio:
-            bf = new.get_integrated_beam_factor(
-                model=freq_model, freqs=data.freqs.to_value("MHz")
-            )
-        else:
-            bf = new.get_mean_beam_factor(
-                model=freq_model, freqs=data.freqs.to_value("MHz")
-            )
-
-        new_data[:, :, i] /= bf
-        if resids is not None:
-            resids[:, :, i] /= bf
+    if use_beam_factor:
+        bf = np.loadtxt(beam_file)
+        new_data *= bf[:, 3]
```

Comment:
You need to also add

```
if resids is not None:
    resids[:, :, i] *= bf
```

You're sure this beam factor should be multiplied, not divided?